-1 
Tested at: 9a6e3dbc9a1f5ded200351613f9ffc8c5b862f47
 I found an error when building: 
 
 <pre>>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HFShowerLibrary.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HFShowerPMT.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HFShowerParam.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HGCNumberingScheme.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HGCNumberingScheme.cc: In member function 'uint32_t HGCNumberingScheme::getUnitID(ForwardSubdetector, int, int, int, int, const G4ThreeVector&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HGCNumberingScheme.cc:48:14: error: 'hgcons' was not declared in this scope
     wafer =  hgcons->waferFromCopy(module);
              ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-21-2300/src/SimG4CMS/Calo/src/HGCNumberingScheme.cc:58:11: error: 'check_' was not declared in this scope
       if (check_) {
           ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13038/10684/summary.html

--- a/Geometry/HGCalCommonData/src/HGCalDDDConstants.cc
+++ b/Geometry/HGCalCommonData/src/HGCalDDDConstants.cc
@@ -22,6 +22,11 @@ HGCalDDDConstants::HGCalDDDConstants(const HGCalParameters* hp,
   mode_ = HGCalGeometryMode( hgpar_->mode_ );
   if (mode_ == HGCalGeometryMode::Square) {
     rmax_ = 0;
+    
+    for( int simreco = 0; simreco < 2; ++simreco ) {
+      tot_layers_[simreco] = layersInit((bool)simreco);
+    }
+
     edm::LogInfo("HGCalGeom") << "HGCalDDDConstants initialized for " << name
 			      << " with " << layers(false) << ":" <<layers(true)
 			      << " layers, " << sectors() << " sectors and "

--- a/SimG4CMS/Calo/src/HGCNumberingScheme.cc
+++ b/SimG4CMS/Calo/src/HGCNumberingScheme.cc
@@ -45,7 +45,7 @@ uint32_t HGCNumberingScheme::getUnitID(ForwardSubdetector subdet, int layer,
       index = 0;
     }
   } else {    
-    wafer =  hgcons->waferFromCopy(module);
+    wafer =  hgcons_.waferFromCopy(module);
     celltyp = cell/1000;
     icell   = cell%1000;
     if (celltyp != 1) celltyp = 0;    
@@ -55,17 +55,16 @@ uint32_t HGCNumberingScheme::getUnitID(ForwardSubdetector subdet, int layer,
     //check if it fits
     if (!hgcons_.isValid(layer,wafer,icell,false)) {
       index = 0;
-      if (check_) {
-        std::pair<int,int> temp = hgcons->assignCell(pos.x(),pos.y(),layer,0,false);
-        std::cout << "wafer from copy: " << wafer << ", from assign: " << temp.first << std::endl;
-        std::cout << "cell from copy:  " << icell << ", from assign: " << temp.second << std::endl;
 
-	edm::LogError("HGCSim") << "[HGCNumberingScheme] ID out of bounds :"
-				<< " Subdet= " << subdet << " Zside= " << iz
-				<< " Layer= " << layer << " Wafer= " << wafer
-				<< " CellType= " << celltyp << " Cell= "
-				<< icell;
-      }    
+      std::pair<int,int> temp = hgcons_.assignCell(pos.x(),pos.y(),layer,0,false);
+      edm::LogError("HGCSim") << "wafer from copy: " << wafer << ", from assign: " << temp.first << std::endl;
+      edm::LogError("HGCSim") << "cell from copy:  " << icell << ", from assign: " << temp.second << std::endl;
+      
+      edm::LogError("HGCSim") << "[HGCNumberingScheme] ID out of bounds :"
+                              << " Subdet= " << subdet << " Zside= " << iz
+                              << " Layer= " << layer << " Wafer= " << wafer
+                              << " CellType= " << celltyp << " Cell= "
+                              << icell;
     }
   }
 #ifdef DebugLog

